Source/Scene/PointCloudEyeDomeLighting.js

```diff
@@ -243,24 +246,45 @@ PointCloudEyeDomeLighting.prototype.update = function (
     ) {
       continue;
     }
-    var derivedCommand = command.derivedCommands.pointCloudProcessor;
+
+    // These variables need to get reset for each iteration. It has to be
+    // done manually since var is function scope not block scope.
+    derivedCommand = undefined;
+    originalShaderProgram = undefined;
+
+    var derivedCommandObject = command.derivedCommands.pointCloudProcessor;
+    if (defined(derivedCommandObject)) {
+      derivedCommand = derivedCommandObject.command;
+      originalShaderProgram = derivedCommandObject.originalShaderProgram;
+    }
+
     if (
       !defined(derivedCommand) ||
       command.dirty ||
       dirty ||
+      originalShaderProgram !== command.shaderProgram ||
       derivedCommand.framebuffer !== this._framebuffer
     ) {
-      // Prevent crash when tiles out-of-view come in-view during context size change
-      derivedCommand = DrawCommand.shallowClone(command);
-      command.derivedCommands.pointCloudProcessor = derivedCommand;
-
+      // Prevent crash when tiles out-of-view come in-view during context size change or
+      // when the underlying shader changes while EDL is disabled
+      derivedCommand = DrawCommand.shallowClone(command, derivedCommand);
       derivedCommand.framebuffer = this._framebuffer;
       derivedCommand.shaderProgram = getECShaderProgram(
         frameState.context,
         command.shaderProgram
       );
       derivedCommand.castShadows = false;
       derivedCommand.receiveShadows = false;
+
+      if (defined(derivedCommandObject)) {
```

Comment:
To make sure I understand this, does the `else` block only get reached on the first frame before `pointCloudProcessor` is given a value?

Also it might not help performance, but would it be more precise to wrap both the `if` and `else` here with an `if (originalShaderProgram !== command.shaderProgram)` so that the `pointCloudProcessor` object only gets updated when necessary? In Sandcastle doing this didn't seem to have any effect (as I'd expect) but maybe it's wrong if it's possible that the command or shader program need to get updated for a reason other than the mismatch we check for..

Reply:
> To make sure I understand this, does the `else` block only get reached on the first frame before `pointCloudProcessor` is given a value?

Yeah that's what's going on. I reorganized the code a bit so it is hopefully more self-documenting.

It should be ok to set `originalShaderProgram` every time even if the command is rederived for some other reason besides shader changes.